Add update functions that run in a transaction.

cdepillabout opened the pull request.

cdepillabout committed:
Add update functions that run in a transaction.

cdepillabout committed:
Fix the documentation.

WraithM commented:
You should rename those `pred` arguments. That shadows a `Prelude` function: https://hackage.haskell.org/package/base-4.8.2.0/docs/Prelude.html#v:pred

WraithM commented:
But I like this other than the `pred` name :) Thanks!

cdepillabout committed:
Renaming pred argument to update functions.

Rename the `pred` argument to the update functions to `predicate`.

`pred` shadows the `pred` function from the Prelude.

cdepillabout commented:
@WraithM Good catch!  Thanks!

I just pushed commit 7f85c62 that renames the `pred` argument to `predicate`.  How does this look?
